index.php

```diff
@@ -634,7 +634,12 @@ public function extractDownload() {
 		$downloadedVersion = $this->getVersionByVersionFile(dirname($downloadedFilePath) . '/nextcloud/version.php');
 		$currentVersion = $this->getVersionByVersionFile($this->baseDir . '/../version.php');
 		if(version_compare($downloadedVersion, $currentVersion, '<')) {
-			throw new \Exception('Downloaded version is lower than installed version');
+			// If the version is not lower, only the same major and minor are allowed
```

Comment:
I guess there it is not that bad, because it had already the same versions migration steps executed and is done because of the daily tar ball (having the same version number quite often).

For the migration oc to nc we should mandatorily execute the migration steps also in those cases, because those are completely different code bases.